src/client.ts

```diff
@@ -199,6 +195,62 @@ export class SymplifySDK {
     }
 }
 
+/**
+ * Calculate the variation allocation for the given visitor ID,
+ * save the allocation in the site data.
+ */
+function allocateVariation(
+    siteData: WebsiteData,
+    project: ProjectConfig,
+    visitorID: string,
+): VariationConfig | null {
+    const variation = findVariationForVisitor(project, visitorID);
+
+    if (variation) {
+        siteData.rememberAllocation(project, variation);
+    } else {
+        siteData.rememberNullAllocation(project);
+    }
+
+    return variation;
+}
+
+/**
+ * Get the preview variation name for the given visitor ID,
+ * update cookies if needed.
+ */
+function handlePreview(
+    siteData: WebsiteData,
+    project: ProjectConfig,
+    cookies: CookieWriter,
+    audienceAttributes: AudienceAttributes,
+    log: Logger,
+): string | null {
+    // While this function looks quite similar to allocating a variation
+    // normally, there are other things we handle here, such as tracing the
+    // audience evaluation.
+
+    const audienceTrace = project.audience?.trace({ attributes: audienceAttributes });
+    if (audienceTrace) {
+        cookies.set("sg_audience_trace", JSON.stringify(audienceTrace), 0);
```

Comment:
This cookie is lost immediately in growers due to this zero.

We will either have to pick a number (say one day), to make it stay around enough for the preview frontend to show it, or make it a session cookie.

To use a session cookie we have to redefine our CookieWriter interface:
* we can make the expireInDays parameter an optional number and define null to mean "session cookie"
* we can define expireInDays=0 to mean "session cookie"
* we can add a new method besides set, e.g. setSession

I think just using one day here is the simplest change, and should not pose any problem. The preview frontend is the only thing that cares about the cookie, the trace will be refreshed on each preview request, and the preview frontend has a cleanup function upon exiting so no old data should linger after it was needed.

Reply:
made it expire in one day